bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairControllerHandler.java

```diff
@@ -0,0 +1,819 @@
+
+    private boolean serviceMode = false;
+    private Unit<Temperature> uom = SIUnits.CELSIUS;
+
+    private final Logger logger = LoggerFactory.getLogger(PentairControllerHandler.class);
+
+    private @Nullable ScheduledFuture<?> syncTimeJob;
+
+    private long lastScheduleTypeWrite;
+
+    private final ExpiringCache<PentairControllerStatus> controllerStatusCache = new ExpiringCache<>(CACHE_EXPIRY);
+    private final ExpiringCache<PentairHeatStatus> heatStatusCache = new ExpiringCache<>(CACHE_EXPIRY);
+
+    private int majorrev, minorrev;
+
+    private PentairControllerActions actions = new PentairControllerActions();
+
+    @SuppressWarnings("unchecked")
+    private final ExpiringCache<PentairControllerCircuit>[] circuitsCache = new ExpiringCache[NUM_CIRCUITS];
+    @SuppressWarnings("unchecked")
+    private final ExpiringCache<PentairControllerSchedule>[] schedulesCache = new ExpiringCache[NUM_SCHEDULES];
+
+    private @Nullable PentairControllerLightMode lightMode;
+
+    public PentairControllerHandler(Thing thing) {
+        super(thing);
+
+        for (int i = 0; i < NUM_SCHEDULES; i++) {
+            schedulesCache[i] = new ExpiringCache<PentairControllerSchedule>(CACHE_EXPIRY_LONG);
+        }
+
+        for (int i = 0; i < NUM_CIRCUITS; i++) {
+            circuitsCache[i] = new ExpiringCache<PentairControllerCircuit>(CACHE_EXPIRY_LONG);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        for (String group : CIRCUIT_GROUPS) {
+            circuitSwitchUIDs.add(new ChannelUID(new ChannelGroupUID(this.getThing().getUID(), group),
+                    CHANNEL_CONTROLLER_CIRCUITSWITCH));
+        }
+
+        super.initialize();
+    }
+
+    @Override
+    public void goOnline() {
+        // Only a single controller is supported on the Pentair bus so prevent multiple controller
+        // things being created.
+        PentairBaseBridgeHandler bridgeHandler = getBridgeHandler();
+
+        if (bridgeHandler == null) { // will not be null here since this is validated in initialize of the super
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.bridge-missing");
+            return;
+        }
+
+        PentairControllerHandler handler = bridgeHandler.findController();
+
+        if (handler != null && !handler.equals(this)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.duplicate-controller");
+        } else {
+            super.goOnline();
+
+        }
+    }
+
+    @Override
+    public void finishOnline() {
+        super.finishOnline();
+        actions.initialize(Objects.requireNonNull(getBridgeHandler()).getBaseActions(), getPentairID());
+
+        // setup syncTimeJob to run once a day, initial time to sync is 3 minutes after controller goes online. This is
```

Comment:
Is this long delay really needed? Have not seen it in other bindings. Or was this tested on a very slow machine?

Reply:
I changed to include as part of the initialization routine.  The 3 minute was just a safe number to use, but putting it in the init routine is a better place.